src/extension/engine/AbstractServer.ts

```diff
@@ -139,6 +139,19 @@ export class AbstractServer
         });
     }
 
+    /**
+     * @param {string} sql - a SQL string
+     * @return {string} - the SQL string without comments
+     */
+    removeComments(sql: string) {
+        const quotes=/^((?:[^"`']*?(?:(?:"(?:[^"]|\\")*?(?<!\\)")|(?:'(?:[^']|\\')*?(?<!\\)')|(?:`(?:[^`]|\\`)*?(?<!\\)`)))*?[^"`']*?)/;
+        const cStyleComments=new RegExp(quotes.source+'/\\*.*?\\*/');
+        const doubleDashComments=new RegExp(quotes.source+'--.*(\r\n|\n|\r)?');
+        while(sql.match(cStyleComments)) sql=sql.replace(cStyleComments,'$1');
+        while(sql.match(doubleDashComments)) sql=sql.replace(doubleDashComments,'$1$2');
```

Comment:
We can replace all occurrences with `global` flag in RegEx. 

```
const doubleDashComments=new RegExp(quotes.source+'--.*(\r\n|\n|\r)?', 'g');
sql=sql.replace(doubleDashComments,'$1$2');
```

is equal 

```
const doubleDashComments=new RegExp(quotes.source+'--.*(\r\n|\n|\r)?');
while(sql.match(doubleDashComments)) sql=sql.replace(doubleDashComments,'$1$2');
```

Reply:
This is a quick test I made in the Node console:
```
> let quotes=/^((?:[^"`']*?(?:(?:"(?:[^"]|\\")*?(?<!\\)")|(?:'(?:[^']|\\')*?(?<!\\)')|(?:`(?:[^`]|\\`)*?(?<!\\)`)))*?[^"`']*?)/;
undefined

> let doubleDashComments=new RegExp(quotes.source+'--.*(\r\n|\n|\r)?');  //without global flag
undefined

> let testString=`SELECT * from table; -- comment
--another comment
SELECT * from table limit 10;`;
undefined

> testString.match(doubleDashComments);
Array(3) ["SELECT * from table; -- comment\n", "SELECT * from table; ", "\n"]

> doubleDashComments=new RegExp(quotes.source+'--.*(\r\n|\n|\r)?', 'g');  //with global flag
/^((?:[^"`']*?(?:(?:"(?:[^"]|\\")*?(?<!\\)")|(?:'(?:[^']|\\')*?(?<!\\)')|(?:`(?:[^`]|\\`)*?(?<!\\)`)))*?[^"`']*?)--.*(\r\n|\n|\r)?/g {lastIndex: 0}

> testString.match(doubleDashComments);
Array(1) ["SELECT * from table; -- comment\n"]
```
Since the regex contains a start of string anchor even with `global` flag set only the first match is returned. Moreover I don't get the capturing groups with the flag set.
This is probably a little time-consuming task but it's the only way I managed to recognize comment patterns outside quotes and remove them.